coord/src/dist_aggregate.c

```diff
@@ -195,35 +199,43 @@ typedef struct {
 } RPNet;
 
 static int getNextReply(RPNet *nc) {
-  while (1) {
-    MRReply *root = MRIterator_Next(nc->it);
-    if (root == MRITERATOR_DONE) {
+  if (nc->cmd.forCursor) {
+    // if there is only 1 reply left (or none), trigger READs at the shards.
+    // TODO: This hardcoded 1 could be replaced with a configurable value (global or query specific)
```

Comment:
I believe we should already replace even just for the case we will need to change it for one customer to whom it will cause some performance degradation.